application-admintools-ui/src/main/resources/AdminTools/Code/AdminToolsJS.xml

```diff
@@ -246,6 +248,25 @@
       $('#downloadFilesModal .download-logs-date-fields').hide();
     }
   });
+
+  // Remove broken dashboard gadget titles, since in some cases these will be either empty, with the wrong title or
+  // will have no ID set, which will make the link unusable. This method should be updated after upgrading to a
+  // XWiki parent that includes fixes for XWIKI-20600: Gadget headings don't have an ID set, or XRENDERING-707: The toc
+  // macro use the name of the page for empty heading entries. We select those headings that contain the element
+  // 'span.wikigeneratedlinkcontent' and remove the text if it has children, or the entire heading otherwise.
```

Comment:
Would be good to mention the reason for targeting `span.wikigeneratedlinkcontent` elements. How exactly are these linked to the problems mentioned above (e.g. XWIKI-20600 and XRENDERING-707)?